src/NuGet.Services.AzureSearch/SearchService/AuxiliaryFileClient.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.WindowsAzure.Storage;
+using Newtonsoft.Json;
+using NuGet.Indexing;
+using NuGetGallery;
+
+namespace NuGet.Services.AzureSearch.SearchService
+{
+    public class AuxiliaryFileClient : IAuxiliaryFileClient
+    {
+        private readonly ICloudBlobClient _cloudBlobClient;
+        private readonly IOptionsSnapshot<SearchServiceConfiguration> _options;
+        private readonly ILogger<AuxiliaryFileClient> _logger;
+        private readonly Lazy<ICloudBlobContainer> _lazyContainer;
+
+        public AuxiliaryFileClient(
+            ICloudBlobClient cloudBlobClient,
+            IOptionsSnapshot<SearchServiceConfiguration> options,
+            ILogger<AuxiliaryFileClient> logger)
+        {
+            _cloudBlobClient = cloudBlobClient ?? throw new ArgumentNullException(nameof(cloudBlobClient));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            _lazyContainer = new Lazy<ICloudBlobContainer>(
```

Comment:
>_lazyContainer [](start = 12, length = 14)

Does this need to be lazy? AFAIK, this just creates some objects (see [this](https://www.fuget.org/packages/WindowsAzure.Storage/9.3.3/lib/netstandard1.3/Microsoft.WindowsAzure.Storage.dll/Microsoft.WindowsAzure.Storage.Blob/CloudBlobClient?code=true#M%3AMicrosoft.WindowsAzure.Storage.Blob.CloudBlobClient.GetContainerReference%28System.String%29)).

Reply:
Yeah, I've investigated the same thing before. I chose this pattern since it was undocumented at the time of my investigation about what kind of validation or "work" is done here. In general, I prefer the debugging experience of failures in method bodies (ASP.NET "actions") instead of in activation (in the ASP.NET pipeline before you enter the constructor).

If you feel strongly I can change this.

Reply:
I pushed this PR but if you want follow-up I can take care of that in a subsequent PR. Just let me know.